application/src/main/java/org/togetherjava/tjbot/commands/help/OnGuildBanDeleteSpamThreadListener.java

```diff
@@ -0,0 +1,49 @@
+package org.togetherjava.tjbot.commands.help;
+
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.guild.GuildBanEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jooq.Result;
+
+import org.togetherjava.tjbot.commands.EventReceiver;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.HelpThreadsRecord;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.togetherjava.tjbot.db.generated.Tables.HELP_THREADS;
+
+/**
+ * Implements the functionality to delete recent help threads when the user gets banned.
+ */
+public final class OnGuildBanDeleteSpamThreadListener extends ListenerAdapter
+        implements EventReceiver {
+    private static final Duration recentThreadDuration = Duration.ofMinutes(30);
+
+    private final Database database;
+
+    /**
+     * @param database the database to get help threads from
+     */
+    public OnGuildBanDeleteSpamThreadListener(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public void onGuildBan(GuildBanEvent event) {
```

Comment:
when an user is banned multiple times (ban, unban, ban), this will fetch the same entries from the DB again. make sure that the code behaves well in such a case.

i suspect it would, right now, send a lot of exceptions.

also, same happens if a thread was already deleted manually by a mod. the thread-database must not be taken as point of truth, its merely a "snapshot" of a single moment. so u must handle it with more care.

i.e. u should add some sort of "if already deleted" check. and also acount for `getThreadChannelById` to not find it etc. and then add maybe a debug-level log to that, but not further exceptions.

Reply:
done